### PR TITLE
Upgrade shorthand type to {type:Type} when flattening. Fixes #4066.

### LINKED ISSUE #4066: Shorthand property types not respected during deserialization in 2.x
### Description
Shorthand property types are not respect during deserialization in 2.x.

```js
// works
properties: {
  longhand: {
    type: Boolean
  }
}
```
```js
// no longer works
properties: {
  shorthand: Boolean
}
```

#### Live Demo

http://jsbin.com/zibapig/edit?html,console

### Versions
- Polymer: v2.x

### PR DESCRIPTION
**Pull request by kevinpschaaf**
### Reference Issue
Fixes #4066

### CODE REVIEW

**Comment by jfrazzano** [comment]
You have the little one yet kev? Congrats!

File: src/elements/element.html [hunk 1]
```diff
@@ -49,10 +60,10 @@
       // (2) element default values
       static get _flattenedProperties() {
         if (!this.hasOwnProperty('__flattenedProperties')) {
-          this.__flattenedProperties = this._ownConfig.properties || {};
+          this.__flattenedProperties = flattenProperties({}, this._ownConfig.properties);
```
Comment (sorvell) [inline src/elements/element.html]: Let's add a comment here that we need to scrutinize this when optimizing. There's an extra loop that we didn't need before and the custom `flattenProperties` prevents us from potentially using `Object.assign`. It's probably nbd but I feel safer putting a bookmark in the code that we made a decision here.

Comment (kevinpschaaf) [inline src/elements/element.html]: Done, added comment.

**Comment by kevinpschaaf** [comment]
Rebasing against `2.0-preview`, which should fix test issues.

**Pull request merged**